### ISSUE TITLE
How to obtain templates with fingerprint samples / features?

### ISSUE THREAD

**Issue by mydoghasworms** [hidglobal/digitalpersona-enrollment#1]
I was previously using the DigitalPersona One Touch SDK API to obtain a fingerprint template.

Now I have tested the @hidglobal/digitalpersona-devices library to obtain fingerprints from a device, but unfortunately that library and this one does not seem to have the functionality to generate a template from a fingerprint feature set.

I also do not see this functionality exposed in the WebSdk.

Is there perhaps something I have missed in the documentation, or perhaps there are undocumented functions to do this in the digitalpersona-enrollment library?

**Comment by a-bronx** [comment]
TL;DR: to extract features and to perform matching you need a fingerprint engine. You can use DigitalPersona Authentication Server which has such engine (FingerJet), or use the same FingerJet engine provided with the DigitalPersona One Touch SDK, or use any other engine of your choice.

The purpose of the `digitalpersona-devices` library is to acquire raw biometric data from fingerprint devices, using JavaScript code running in web browsers, and a help of the DigitalPersona Agent (a native app in DP Workstation or Lite Client). The library does not generate fingerprint templates, because it requires a fingerprint engine which is not a part of DP Workstation or Lite Client.

When raw fingerprint data is acquired, you use the `digitalpersona-enrollment` library to send the collected fingerprint images to the DigitalPersona Authentication Server for enrollment. The DP Server has a fingerprint engine which creates fingerprint templates during enrollment, and stores them in a cryptographically protected storage. Templates never leave the server, for security purposes. You cannot use the library or WebSdk to obtain templates.

After enrollment, users can authenticate with their fingerprints using the `digitalpersona-authentication` library and the DP Server, which uses its engine for fingerprint matching.

**Comment by mydoghasworms** [comment]
@a-bronx Thanks for the explanation. I feared as much.

I think I will end up using @hidglobal/digitalpersona-devices to capture fingerprints from the device and building a custom service using the One Touch SDK for creating templates from fingerprint samples, as my client has a custom system where they want to store the fingerprints.
